component-samples/aio/src/main/java/org/fidoalliance/fdo/sample/AioServerApp.java

```diff
@@ -0,0 +1,279 @@
+    wrapper = tomcat.addServlet(ctx, "replacementVoucherServlet",
+        new OwnerReplacementVoucherServlet());
+    wrapper.addMapping("/api/v1/owner/newvoucher/*");
+    wrapper = tomcat.addServlet(ctx, "setupinfoServlet",
+        new OwnerSetupInfoServlet());
+    wrapper.addMapping("/api/v1/owner/setupinfo/*");
+    wrapper.setAsyncSupported(true);
+    wrapper = tomcat.addServlet(ctx, "ownerCustomerServlet",
+        new OwnerCustomerServlet());
+    wrapper.addMapping("/api/v1/owner/customer/*");
+
+    wrapper = tomcat.addServlet(ctx, "SystemResource", new OwnerSystemResourceServlet());
+    wrapper.setAsyncSupported(true);
+    wrapper.addMapping("/api/v1/device/svi");
+
+    wrapper = tomcat.addServlet(ctx, "sviSettingsServlet",
+        new OwnerSviSettingsServlet());
+    wrapper.addMapping("/api/v1/owner/svi/settings/*");
+
+    wrapper = tomcat.addServlet(ctx, "DiApi", new DiApiServlet());
+    wrapper.addMapping("/api/v1/vouchers/*");
+    wrapper.setAsyncSupported(true);
+
+    wrapper = tomcat.addServlet(ctx, "AssignCustomerApi", new AssignCustomerServlet());
+    wrapper.addMapping("/api/v1/assign/*");
+    wrapper.setAsyncSupported(true);
+
+    wrapper = tomcat.addServlet(ctx, "UpdateRvInfoApi", new RvInfoServlet());
+    wrapper.addMapping("/api/v1/rvinfo/*");
+    wrapper.setAsyncSupported(true);
+    //wrapper.setLoadOnStartup(4);
+
+    wrapper = tomcat.addServlet(ctx, "downloadServlet", new AioFileDownloadServlet());
+    wrapper.addMapping("/downloads/*");
+    wrapper.addInitParameter(AioAppSettings.DOWNLOADS_PATH,
+        AioConfigLoader.loadConfig(AioAppSettings.DOWNLOADS_PATH));
+    wrapper.setAsyncSupported(true);
+
+    wrapper = tomcat.addServlet(ctx, "uploadServlet", new AioFileUploadServlet());
+    wrapper.addMapping("/api/v1/uploads/*");
+    wrapper.addInitParameter(AioAppSettings.DOWNLOADS_PATH,
+        AioConfigLoader.loadConfig(AioAppSettings.DOWNLOADS_PATH));
+    wrapper.setAsyncSupported(true);
+
+
+    wrapper = tomcat.addServlet(ctx, "H2Console", new WebServlet());
+    wrapper.addMapping("/console/*");
+    // To enable remote connections to the DB set webAllowOthers=true
+    // This creates a security hole in the system.
+    // Not recommended to use especially on production system
+    wrapper.addInitParameter("webAllowOthers",
+        AioConfigLoader.loadConfig(AioAppSettings.DB_ALLOW_OTHERS));
+    //wrapper.addInitParameter("trace", "");
+    wrapper.setLoadOnStartup(3);
+
+
+    //setup digest auth
+    LoginConfig config = new LoginConfig();
+    config.setAuthMethod(AioAppSettings.AUTH_METHOD);
+    ctx.setLoginConfig(config);
+    ctx.addSecurityRole(AioAppSettings.AUTH_ROLE);
+    SecurityConstraint constraint = new SecurityConstraint();
+    constraint.addAuthRole(AioAppSettings.AUTH_ROLE);
+    SecurityCollection collection = new SecurityCollection();
+    collection.addPattern("/api/v1/*");
+    constraint.addCollection(collection);
```

Comment:
collection.addPattern("/downloads/*") missing

Reply:
"/downloads/*" is intentionally missing because its the only API that  does not require any authentication.  Its for downloading agents/scripts that have no secrets in them.

all api's  should be covered by /api/v1/*    all servlet api patterns are "/api/v1/*" e.g. /api/v1/device/svi""